src/modules/mavlink/mavlink_main.cpp

```diff
@@ -2156,6 +2079,13 @@ Mavlink::task_main(int argc, char *argv[])
 
 #endif // MAVLINK_UDP
 
+	if (!set_instance_id()) {
```

Comment:
I haven't checked in detail, but I believe this needs to be after `configure_streams_to_default`